src/kernels/cosine.jl

```diff
@@ -0,0 +1,14 @@
+"""
+    CosineKernel
+
+The cosine kernel is a stationary kernel for a sinusoidal given by
+```
+    κ(x,y) = cos( 2π * (x-y) )
+```
+
+"""
+struct CosineKernel <: BaseKernel end
+
+kappa(κ::CosineKernel, d::Real) = cos(2*pi*d)
```

Comment:
Was just browsing through the Julia documentation and noticed that there is a function `cospi` that computes `cos(pi*x)` more accurately than the "naive" implementation, particularly for large `x`. I guess we could use it here?
```suggestion
kappa(κ::CosineKernel, d::Real) = cospi(2*d)
```
I'm also wondering if we should keep the scaling with 2 here - I don't know what's commonly used since the documentation of GPML (no scaling) and its implementation (scaling) are not consistent in that point and GPyTorch does not scale by a factor of 2 (according to the docs). I guess that decision could be based on the default domain that we want to support?

Reply:
I am also wondering more generally about the validity of this kernel. cf this question : https://www.quora.com/Is-cosine-kernel-as-given-in-Kernel-statistics-a-radial-basis-function-kernel
I think it is indeed only valid when whatever is inside `cos` is determined between -pi/2 and pi/2. Is it really psd if inputs are outside of this?

Reply:
Also noted here : https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use
Apparently it should return 0 outside

Reply:
Oh I just realised that these kernels are more meant for convolution stuff. Nevermind

Reply:
Does it help to stay consistent with `GPyTorch`?

Reply:
Not really, every package has its own convention... I like pi/2 because then for a distance of 1 the kernel outputs 0